src/nectarchain/dqm/dqm_summary_processor.py

```diff
@@ -40,50 +40,39 @@ def WriteAllResults(self, path, DICT):
         hdu, hdu0, hdu1, hdu2 = None, None, None, None
         hdulist = fits.HDUList()
         for i, j in DICT.items():
-            if (i == "Results_TriggerStatistics"):
+            if i == "Results_TriggerStatistics":
                 for n2, m2 in j.items():
                     data2[n2] = m2
-                hdu2 = fits.BinTableHDU(data2)
-                hdu2.name = "Trigger"
+                    hdu2 = fits.BinTableHDU(data2)
+                    hdu2.name = n2
+                    hdulist.append(hdu2)
 
             elif (i == "Results_MeanWaveForms_HighGain") or (
                 i == "Results_MeanWaveForms_LowGain"
             ):
                 for n1, m1 in j.items():
                     data1[n1] = m1
-                hdu1 = fits.BinTableHDU(data1)
-                hdu1.name = "MWF"    
+                    hdu1 = fits.BinTableHDU(data1)
+                    hdu1.name = n1
+                    hdulist.append(hdu1)
 
-            elif (i == "Results_PixelTimeline_HighGain") or (i == "Results_PixelTimeline_LowGain"):
+            elif (i == "Results_PixelTimeline_HighGain") or (
+                i == "Results_PixelTimeline_LowGain"
+            ):
                 for n0, m0 in j.items():
-                    data0[n0] = m0 
-                hdu0 = fits.BinTableHDU(data0)
-                hdu0.name = "BPX"
+                    data0[n0] = m0
+                    hdu0 = fits.BinTableHDU(data0)
+                    hdulist.append(hdu0)
```

Comment:
The name is not set, is that normal ? 
I would expect to see the line : `hdu0.name = n0` or something like that.

Reply:
Ah, you're absolutely right.
And I think I found a way to simplify this method, see last commit.

Reply:
Looks much better indeed. 

The change imply that now you have one `BinTableHDU` per result instead of 2 informations in one as I understand the code for some of the if cases.  Some names will also changed as some were hard coded in the previous version. Will it be a problems for code coming after the creation of the fits ?

Reply:
Actually, yes, it will. The DQM itself fails with such output structure. I should revert this.

Reply:
It's weird because the "else" part of the old code is doing what you are doing in the "new" static function you are using. Maybe it should not be reverted but things can be fixed ? (I find more appealing to not have the long list of if/else :-) )

Reply:
Hi @vmarandon ,
I am currently working on that. The current error reads:
```python
Traceback (most recent call last):
  File "/opt/cta/nectarchain/src/nectarchain/dqm/start_dqm.py", line 243, in <module>
    p.WriteAllResults(ResPath, NESTED_DICT)
  File "/opt/cta/nectarchain/src/nectarchain/dqm/dqm_summary_processor.py", line 49, in WriteAllResults
    hdu = self._create_hdu(name, content)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/cta/nectarchain/src/nectarchain/dqm/dqm_summary_processor.py", line 40, in _create_hdu
    data[name] = content
    ~~~~^^^^^^
  File "/opt/conda/envs/nectar-dev/lib/python3.11/site-packages/astropy/table/table.py", line 2092, in __setitem__
    self.add_column(value, name=item, copy=True)
  File "/opt/conda/envs/nectar-dev/lib/python3.11/site-packages/astropy/table/table.py", line 2395, in add_column
    raise TypeError("Empty table cannot have column set to scalar value")
TypeError: Empty table cannot have column set to scalar value
```
as we are hitting something like "mixin" columns in `astropy.table`: https://docs.astropy.org/en/stable/table/mixin_columns.html#registering-array-like-objects-as-mixin-columns. I am looking for a solution avoiding the need to implement such a wrapper class.

Reply:
This boils down to the fact that the `TRIGGER-STATISTICS` and `START-TIMES` fields are just strings, and as such, cannot be stored as columns in `astropy.table`. The commit 95491ab324e9ea9432324471ada88c58715a56ac worked, but at the expanse of duplicated columns in the FITS output, as we noted with @hashkar.
Avoiding this duplication in a4dd3c47dcc8571dfd4a2d3c305855c2dd9974c0 triggered the issue discussed here.
I would see two solutions:
- simply ignoring the `TRIGGER-STATISTICS` and `START-TIMES` fields, which is not elegant, with something like:
```python
    def WriteAllResults(self, path, DICT):
        hdulist = fits.HDUList()
        for i, j in DICT.items():
            for name, content in j.items():
                try:
                    hdu = self._create_hdu(name, content)
                    hdulist.append(hdu)
                except TypeError as e:
                    print(f"DEBUG JPL: Got error {e}, skipping {name}")
                    pass
```
- or changing their format and storing them as dicts in https://github.com/cta-observatory/nectarchain/blob/main/src/nectarchain/dqm/trigger_statistics.py.

I'd go for the 2nd one.

Reply:
... which gives the same issue...

Reply:
OK, I think I found a way.